app/domain/services/fetch_course_events/service.rb

```diff
@@ -131,21 +131,30 @@ def fetch_and_process_course_events(courses, event_types, restart, current_time)
       course_uuids_to_requery << course.uuid \
         unless course_event_response.fetch(:is_gap) || course_event_response.fetch(:is_end)
 
+      limit_sequence_number = [
+        course.sequence_number, events.map { |event| event.fetch(:sequence_number) }.max + 1
+      ].max
+
       events_by_type = events.group_by { |event| event.fetch(:event_type) }
```

Comment:
you could use `max_by` here https://ruby-doc.org/core-2.6.5/Enumerable.html#method-i-max_by

Reply:
I would have to call fetch twice with `max_by`... I think map/max is a little more concise here.
One interesting alternative would be `([ course.sequence_number ] + events.map { |event| event.fetch(:sequence_number) + 1 }).max`... hmm maybe I'll do that...